discovery-provider/src/utils/get_all_other_nodes.py

```diff
@@ -96,3 +98,18 @@ async def fetch_results():
             )
 
     return all_other_nodes, all_other_wallets
+
+
+def get_all_other_nodes_cached(redis) -> Tuple[List[str], List[str]]:
```

Comment:
this has the side effect of being slow once in a while. maybe better is to return a value first and then re-evaluate the cache?

state-with-reevaluate kind of thing?

not blocking though

Reply:
you mean on a query re-evaluate in another process? not sure I understand